src/serializer.ts

```diff
@@ -84,7 +84,7 @@ export class Serializer {
      * @param clazz The class constructor.
      * @returns An instance of the type `T` with the prototype of `clazz` or one of its registered children.
      */
-    public deserialize<T>(obj: any, clazz?: any): T {
+    public deserialize<T>(obj: any, clazz?: any /* TODO See if we can change this any */): T {
```

Comment:
I know, that's why I didn't type hint it as Instantiable and just add a TODO for when something come up. See https://github.com/Microsoft/TypeScript/issues/17572